src/Solcore/Desugarer/EmitCore.hs

```diff
@@ -69,6 +67,17 @@ type DataTable = Map.Map Name DataTy
 type TConInfo = ([Tyvar], [DConInfo]) -- type con: type vars and data cons
 type DConInfo = (Name, [Ty])          -- data con: name and argument types
 
+sumDataTy :: DataTy
+sumDataTy = DataTy
+  { dataName = "sum"
+  , dataParams = [TVar "a", TVar "b"]
+  , dataConstrs = [ Constr "inl" [tyvar "a"]
+                  , Constr "inr" [tyvar "b"]
+                  ]
+  } where
+    tyvar = TyVar . TVar
+
+builtinDataInfo = [ ("sum", sumDataTy) ]
```

Comment:
should there also be a `pair` here?

Reply:
> Do you think it's worthwhile to try to remove the special cases?

Definitely not in this pr. I don't have enough context to have a strong opinion on that in general. Maybe lets open an issue so we don't forget?